crates/nu-cli/src/completions/custom_completions.rs

```diff
@@ -240,7 +240,17 @@ impl<'a> Completer for CommandWideCompletion<'a> {
             new_span = Span::new(span.start, span.end.saturating_sub(1));
         }
 
-        let block = working_set.get_block(self.block_id);
+        let mut block = working_set.get_block(self.block_id).clone();
+
+        // LSP completion where custom def is parsed but not compiled
+        if block.ir_block.is_none()
+            && let Ok(ir_block) = compile(working_set, &block)
+        {
+            let mut new_block = (*block).clone();
+            new_block.ir_block = Some(ir_block);
+            block = Arc::new(new_block);
+        }
```

Comment:
@Bahex not sure if there's a better way for this, or maybe we don't need this for lsp completion at all?

The extra cost isn't too bad though.